src/oauth-provider.ts

```diff
@@ -671,26 +689,68 @@ class OAuthProviderImpl {
   /**
    * Represents the validated type of a handler (ExportedHandler or WorkerEntrypoint)
    */
-  private typedApiHandler: TypedHandler;
   private typedDefaultHandler: TypedHandler;
+  
+  /**
+   * Map of API routes to their validated handlers
+   * In the simple case, this will be a single entry with the route and handler from options.apiRoute/apiHandler
+   * In the advanced case, this will contain entries from options.apiHandlers
+   */
+  private typedApiHandlers: Map<string, TypedHandler>;
```

Comment:
It looks like we never actually do a lookup in this Map, we always iterate over it. So why use a map at all? An array of tuples would be more efficient, since they wouldn't need to maintain a hash table.

Reply:
Fixed.